### ISSUE TITLE
Opening lua file problem while highlight enabled. Python is fine, not lua.

### ISSUE THREAD

**Issue by leet0rz** [nvim-treesitter/nvim-treesitter#5224]
### Describe the bug

Not even sure anymore what is causing this error.
The main error message when I open lua files:

```
Error detected while processing BufNewFile Autocommands for "*":
Error executing lua callback: C:\Program Files\Neovim\share\nvim\runtime\filetype.lua:21: Error executing lua: C:\Program Files\Neovim\share\nvim\runtime\filetype.lua:22: BufNewFile Autocommands for "*"..FileType Autocommands for "*": Vim(append):Error executing lua callback: Failed to load parser: uv_dlsym: The specified procedure could not be found.

stack traceback:
        [C]: in function '_ts_add_language'
        ...eovim/share/nvim/runtime/lua/vim/treesitter/language.lua:104: in function 'add'
        ...m/share/nvim/runtime/lua/vim/treesitter/languagetree.lua:109: in function 'new'
        ...m Files/Neovim/share/nvim/runtime/lua/vim/treesitter.lua:64: in function '_create_parser'
        ...m Files/Neovim/share/nvim/runtime/lua/vim/treesitter.lua:130: in function 'get_parser'
        ...m Files/Neovim/share/nvim/runtime/lua/vim/treesitter.lua:457: in function 'start'
        ...a/lazy/nvim-treesitter/lua/nvim-treesitter/highlight.lua:20: in function 'attach'
        ...ata/lazy/nvim-treesitter/lua/nvim-treesitter/configs.lua:509: in function 'attach_module'
        ...ata/lazy/nvim-treesitter/lua/nvim-treesitter/configs.lua:532: in function 'reattach_module'
        ...ata/lazy/nvim-treesitter/lua/nvim-treesitter/configs.lua:133: in function <...ata/lazy/nvim-treesitter/lua/nvim-treesitter/configs.lua:132>
        [C]: in function 'nvim_cmd'
        C:\Program Files\Neovim\share\nvim\runtime\filetype.lua:22: in function <C:\Program Files\Neovim\share\nvim\runtime\filetype.lua:21>
        [C]: in function 'nvim_buf_call'
        C:\Program Files\Neovim\share\nvim\runtime\filetype.lua:21: in function <C:\Program Files\Neovim\share\nvim\runtime\filetype.lua:10>
stack traceback:
        [C]: in function 'nvim_cmd'
        C:\Program Files\Neovim\share\nvim\runtime\filetype.lua:22: in function <C:\Program Files\Neovim\share\nvim\runtime\filetype.lua:21>
        [C]: in function 'nvim_buf_call'
        C:\Program Files\Neovim\share\nvim\runtime\filetype.lua:21: in function <C:\Program Files\Neovim\share\nvim\runtime\filetype.lua:10>
stack traceback:
        [C]: in function 'nvim_buf_call'
        C:\Program Files\Neovim\share\nvim\runtime\filetype.lua:21: in function <C:\Program Files\Neovim\share\nvim\runtime\filetype.lua:10>
```

### To Reproduce

Install treesitter, copy the default config into my plugins.lua then it all came up.

### Expected behavior

syntax highlighting in lua files and no errors.

### Output of `:checkhealth nvim-treesitter`

```markdown
==============================================================================
nvim-treesitter: require("nvim-treesitter.health").check()

Installation ~
- WARNING `tree-sitter` executable not found (parser generator, only needed for :TSInstallFromGrammar, not required for :TSInstall)
- OK `node` found v18.17.0 (only needed for :TSInstallFromGrammar)
- OK `git` executable found.
- OK `gcc` executable found. Selected from { vim.NIL, "cc", "gcc", "clang", "cl", "zig" }
  Version: gcc (MinGW.org GCC-6.3.0-1) 6.3.0
- OK Neovim was compiled with tree-sitter runtime ABI version 14 (required >=13). Parsers must be compatible with runtime ABI.

OS Info:
{
  machine = "x86_64",
  release = "10.0.19044",
  sysname = "Windows_NT",
  version = "Windows 10 Pro"
} ~

Parser/Features         H L F I J
  - lua                 x x x x x

  Legend: H[ighlight], L[ocals], F[olds], I[ndents], In[j]ections
         +) multiple parsers found, only one will be used
         x) errors found in the query, try to run :TSUpdate {lang} ~

The following errors have been detected: ~
- ERROR lua(highlights): Failed to load parser: uv_dlsym: The specified procedure could not be found.
  
  lua(highlights) is concatenated from the following files:
  | [ERROR]:"C:\Users\waffle\AppData\Local\nvim-data\lazy\nvim-treesitter\queries\lua\highlights.scm", failed to load: Failed to load parser: uv_dlsym: The specified procedure could not be found.
  
- ERROR lua(locals): Failed to load parser: uv_dlsym: The specified procedure could not be found.
  
  lua(locals) is concatenated from the following files:
  | [ERROR]:"C:\Users\waffle\AppData\Local\nvim-data\lazy\nvim-treesitter\queries\lua\locals.scm", failed to load: Failed to load parser: uv_dlsym: The specified procedure could not be found.
  
- ERROR lua(folds): Failed to load parser: uv_dlsym: The specified procedure could not be found.
  
  lua(folds) is concatenated from the following files:
  | [ERROR]:"C:\Users\waffle\AppData\Local\nvim-data\lazy\nvim-treesitter\queries\lua\folds.scm", failed to load: Failed to load parser: uv_dlsym: The specified procedure could not be found.
  
- ERROR lua(indents): Failed to load parser: uv_dlsym: The specified procedure could not be found.
  
  lua(indents) is concatenated from the following files:
  | [ERROR]:"C:\Users\waffle\AppData\Local\nvim-data\lazy\nvim-treesitter\queries\lua\indents.scm", failed to load: Failed to load parser: uv_dlsym: The specified procedure could not be found.
  
- ERROR lua(injections): Failed to load parser: uv_dlsym: The specified procedure could not be found.
  
  lua(injections) is concatenated from the following files:
  | [ERROR]:"C:\Users\waffle\AppData\Local\nvim-data\lazy\nvim-treesitter\queries\lua\injections.scm", failed to load: Failed to load parser: uv_dlsym: The specified procedure could not be found.
```

### Output of `nvim --version`

```text
NVIM v0.10.0-dev-798+g72e619ca9
Build type: RelWithDebInfo
LuaJIT 2.1.0-beta3
Run "nvim -V1 -v" for more info
```

**Comment by lucario387** [comment]
What's the config then? I cannot reproduce this. From what I can tell the parser failed to be downloaded, again.

**Comment by leet0rz** [comment]
Config is just the copy paste and I've been messing with cancling git as stated and adding clang as recommended from the troubleshooting:
`require('nvim-treesitter.install').compilers = { "clang" }`

Check health after that when run as admin:
`- WARNING `tree-sitter` executable not found (parser generator, only needed for :TSInstallFromGrammar, not required for :TSInstall)
- OK `node` found v18.17.0 (only needed for :TSInstallFromGrammar)
- OK `git` executable found.
- ERROR `cc` executable not found.
  - ADVICE:
    - Check that any of { "clang" } is in your $PATH or set the environment variable CC or `require"nvim-treesitter.install".compilers` explicitly!
- OK Neovim was compiled with tree-sitter runtime ABI version 14 (required >=13). Parsers must be compatible with runtime ABI.

OS Info:
{
  machine = "x86_64",
  release = "10.0.19044",
  sysname = "Windows_NT",
  version = "Windows 10 Pro"
} ~

Parser/Features         H L F I J
  - lua                 x x x x x

  Legend: H[ighlight], L[ocals], F[olds], I[ndents], In[j]ections
         +) multiple parsers found, only one will be used
         x) errors found in the query, try to run :TSUpdate {lang} ~

The following errors have been detected: ~
- ERROR lua(highlights): Failed to load parser: uv_dlsym: The specified procedure could not be found.
  
  lua(highlights) is concatenated from the following files:
  | [ERROR]:"C:\Users\waffle\AppData\Local\nvim-data\lazy\nvim-treesitter\queries\lua\highlights.scm", failed to load: Failed to load parser: uv_dlsym: The specified procedure could not be found.
  
- ERROR lua(locals): Failed to load parser: uv_dlsym: The specified procedure could not be found.
  
  lua(locals) is concatenated from the following files:
  | [ERROR]:"C:\Users\waffle\AppData\Local\nvim-data\lazy\nvim-treesitter\queries\lua\locals.scm", failed to load: Failed to load parser: uv_dlsym: The specified procedure could not be found.
  
- ERROR lua(folds): Failed to load parser: uv_dlsym: The specified procedure could not be found.
  
  lua(folds) is concatenated from the following files:
  | [ERROR]:"C:\Users\waffle\AppData\Local\nvim-data\lazy\nvim-treesitter\queries\lua\folds.scm", failed to load: Failed to load parser: uv_dlsym: The specified procedure could not be found.
  
- ERROR lua(indents): Failed to load parser: uv_dlsym: The specified procedure could not be found.
  
  lua(indents) is concatenated from the following files:
  | [ERROR]:"C:\Users\waffle\AppData\Local\nvim-data\lazy\nvim-treesitter\queries\lua\indents.scm", failed to load: Failed to load parser: uv_dlsym: The specified procedure could not be found.
  
- ERROR lua(injections): Failed to load parser: uv_dlsym: The specified procedure could not be found.
  
  lua(injections) is concatenated from the following files:
  | [ERROR]:"C:\Users\waffle\AppData\Local\nvim-data\lazy\nvim-treesitter\queries\lua\injections.scm", failed to load: Failed to load parser: uv_dlsym: The specified procedure could not be found.
  `

when I try to TSInstall c it gives me: No C compiler found! "clang" are not executable

Clang is definitely in path, I am running it from my shell. It just depends if I run as admin or not but I've tried both and they both give errors.

Not as admin when launching nvim:

`Error detected while processing BufNewFile Autocommands for "*":
Error executing lua callback: C:\Program Files\Neovim\share\nvim\runtime\filetype.lua:21: Error executing lua: C:\Program Files\Neovim\share\nvim\runtime\filetype.lua:22: BufNewFile Autocommands for "*"..FileType Autocommands for "*": Vim(append):Error executing lua callback: Failed to load parser: uv_dlsym: The specified procedure could not be found.

stack traceback:
        [C]: in function '_ts_add_language'
        ...eovim/share/nvim/runtime/lua/vim/treesitter/language.lua:104: in function 'add'
        ...m/share/nvim/runtime/lua/vim/treesitter/languagetree.lua:109: in function 'new'
        ...m Files/Neovim/share/nvim/runtime/lua/vim/treesitter.lua:64: in function '_create_parser'
        ...m Files/Neovim/share/nvim/runtime/lua/vim/treesitter.lua:130: in function 'get_parser'
        ...m Files/Neovim/share/nvim/runtime/lua/vim/treesitter.lua:457: in function 'start'
        ...a/lazy/nvim-treesitter/lua/nvim-treesitter/highlight.lua:20: in function 'attach'
        ...ata/lazy/nvim-treesitter/lua/nvim-treesitter/configs.lua:509: in function 'attach_module'
        ...ata/lazy/nvim-treesitter/lua/nvim-treesitter/configs.lua:532: in function 'reattach_module'
        ...ata/lazy/nvim-treesitter/lua/nvim-treesitter/configs.lua:133: in function <...ata/lazy/nvim-treesitter/lua/nvim-treesitter/configs.lua:132>
        [C]: in function 'nvim_cmd'
        C:\Program Files\Neovim\share\nvim\runtime\filetype.lua:22: in function <C:\Program Files\Neovim\share\nvim\runtime\filetype.lua:21>
        [C]: in function 'nvim_buf_call'
        C:\Program Files\Neovim\share\nvim\runtime\filetype.lua:21: in function <C:\Program Files\Neovim\share\nvim\runtime\filetype.lua:10>
stack traceback:
        [C]: in function 'nvim_cmd'
        C:\Program Files\Neovim\share\nvim\runtime\filetype.lua:22: in function <C:\Program Files\Neovim\share\nvim\runtime\filetype.lua:21>
        [C]: in function 'nvim_buf_call'
        C:\Program Files\Neovim\share\nvim\runtime\filetype.lua:21: in function <C:\Program Files\Neovim\share\nvim\runtime\filetype.lua:10>
stack traceback:
        [C]: in function 'nvim_buf_call'
        C:\Program Files\Neovim\share\nvim\runtime\filetype.lua:21: in function <C:\Program Files\Neovim\share\nvim\runtime\filetype.lua:10>`

check health when not as admin:
`nvim-treesitter: require("nvim-treesitter.health").check()

Installation ~
- OK `tree-sitter` found 0.20.8 (0c49d6745b3fc4822ab02e0018770cd6383a779c) (parser generator, only needed for :TSInstallFromGrammar)
- OK `node` found v18.17.0 (only needed for :TSInstallFromGrammar)
- OK `git` executable found.
- OK `clang` executable found. Selected from { "clang" }
  Version: clang version 16.0.6
- OK Neovim was compiled with tree-sitter runtime ABI version 14 (required >=13). Parsers must be compatible with runtime ABI.

OS Info:
{
  machine = "x86_64",
  release = "10.0.19044",
  sysname = "Windows_NT",
  version = "Windows 10 Pro"
} ~

Parser/Features         H L F I J
  - lua                 x x x x x

  Legend: H[ighlight], L[ocals], F[olds], I[ndents], In[j]ections
         +) multiple parsers found, only one will be used
         x) errors found in the query, try to run :TSUpdate {lang} ~

The following errors have been detected: ~
- ERROR lua(highlights): Failed to load parser: uv_dlsym: The specified procedure could not be found.
  
  lua(highlights) is concatenated from the following files:
  | [ERROR]:"C:\Users\waffle\AppData\Local\nvim-data\lazy\nvim-treesitter\queries\lua\highlights.scm", failed to load: Failed to load parser: uv_dlsym: The specified procedure could not be found.
  
- ERROR lua(locals): Failed to load parser: uv_dlsym: The specified procedure could not be found.
  
  lua(locals) is concatenated from the following files:
  | [ERROR]:"C:\Users\waffle\AppData\Local\nvim-data\lazy\nvim-treesitter\queries\lua\locals.scm", failed to load: Failed to load parser: uv_dlsym: The specified procedure could not be found.
  
- ERROR lua(folds): Failed to load parser: uv_dlsym: The specified procedure could not be found.
  
  lua(folds) is concatenated from the following files:
  | [ERROR]:"C:\Users\waffle\AppData\Local\nvim-data\lazy\nvim-treesitter\queries\lua\folds.scm", failed to load: Failed to load parser: uv_dlsym: The specified procedure could not be found.
  
- ERROR lua(indents): Failed to load parser: uv_dlsym: The specified procedure could not be found.
  
  lua(indents) is concatenated from the following files:
  | [ERROR]:"C:\Users\waffle\AppData\Local\nvim-data\lazy\nvim-treesitter\queries\lua\indents.scm", failed to load: Failed to load parser: uv_dlsym: The specified procedure could not be found.
  
- ERROR lua(injections): Failed to load parser: uv_dlsym: The specified procedure could not be found.
  
  lua(injections) is concatenated from the following files:
  | [ERROR]:"C:\Users\waffle\AppData\Local\nvim-data\lazy\nvim-treesitter\queries\lua\injections.scm", failed to load: Failed to load parser: uv_dlsym: The specified procedure could not be found.
  
`

**Comment by lucario387** [comment]
What about using zig, mingw_x64 (note, not the 32-bit version of mingw) as compilers?

**Comment by leet0rz** [comment]
With zig on opening lua file:
```
Error detected while processing BufNewFile Autocommands for "*":
Error executing lua callback: C:\Program Files\Neovim\share\nvim\runtime\filetype.lua:21: Error executing lua: C:\Program Files\Neovim\share\nvim\runtime\filetype.lua:22: BufNewFile Autocommands for "*"..FileType Autocommands for "*": Vim(append):Error executing lua callback: Failed to load parser: uv_dlsym: The specified procedure could not be found.

stack traceback:
        [C]: in function '_ts_add_language'
        ...eovim/share/nvim/runtime/lua/vim/treesitter/language.lua:104: in function 'add'
        ...m/share/nvim/runtime/lua/vim/treesitter/languagetree.lua:109: in function 'new'
        ...m Files/Neovim/share/nvim/runtime/lua/vim/treesitter.lua:64: in function '_create_parser'
        ...m Files/Neovim/share/nvim/runtime/lua/vim/treesitter.lua:130: in function 'get_parser'
        ...m Files/Neovim/share/nvim/runtime/lua/vim/treesitter.lua:457: in function 'start'
        ...a/lazy/nvim-treesitter/lua/nvim-treesitter/highlight.lua:20: in function 'attach'
        ...ata/lazy/nvim-treesitter/lua/nvim-treesitter/configs.lua:509: in function 'attach_module'
        ...ata/lazy/nvim-treesitter/lua/nvim-treesitter/configs.lua:532: in function 'reattach_module'
        ...ata/lazy/nvim-treesitter/lua/nvim-treesitter/configs.lua:133: in function <...ata/lazy/nvim-treesitter/lua/nvim-treesitter/configs.lua:132>
        [C]: in function 'nvim_cmd'
        C:\Program Files\Neovim\share\nvim\runtime\filetype.lua:22: in function <C:\Program Files\Neovim\share\nvim\runtime\filetype.lua:21>
        [C]: in function 'nvim_buf_call'
        C:\Program Files\Neovim\share\nvim\runtime\filetype.lua:21: in function <C:\Program Files\Neovim\share\nvim\runtime\filetype.lua:10>
stack traceback:
        [C]: in function 'nvim_cmd'
        C:\Program Files\Neovim\share\nvim\runtime\filetype.lua:22: in function <C:\Program Files\Neovim\share\nvim\runtime\filetype.lua:21>
        [C]: in function 'nvim_buf_call'
        C:\Program Files\Neovim\share\nvim\runtime\filetype.lua:21: in function <C:\Program Files\Neovim\share\nvim\runtime\filetype.lua:10>
stack traceback:
        [C]: in function 'nvim_buf_call'
        C:\Program Files\Neovim\share\nvim\runtime\filetype.lua:21: in function <C:\Program Files\Neovim\share\nvim\runtime\filetype.lua:10>
```

CH:
```
nvim-treesitter: require("nvim-treesitter.health").check()

Installation ~
- OK `tree-sitter` found 0.20.8 (0c49d6745b3fc4822ab02e0018770cd6383a779c) (parser generator, only needed for :TSInstallFromGrammar)
- OK `node` found v18.17.0 (only needed for :TSInstallFromGrammar)
- OK `git` executable found.
- OK `zig` executable found. Selected from { "zig" }
  Version: info: Usage: zig [command] [options]
- OK Neovim was compiled with tree-sitter runtime ABI version 14 (required >=13). Parsers must be compatible with runtime ABI.

OS Info:
{
  machine = "x86_64",
  release = "10.0.19044",
  sysname = "Windows_NT",
  version = "Windows 10 Pro"
} ~

Parser/Features         H L F I J
  - c                   ✓ ✓ ✓ ✓ ✓
  - lua                 x x x x x
  - query               ✓ ✓ ✓ ✓ ✓
  - vim                 ✓ ✓ ✓ . ✓
  - vimdoc              ✓ . . . ✓

  Legend: H[ighlight], L[ocals], F[olds], I[ndents], In[j]ections
         +) multiple parsers found, only one will be used
         x) errors found in the query, try to run :TSUpdate {lang} ~

The following errors have been detected: ~
- ERROR lua(highlights): Failed to load parser: uv_dlsym: The specified procedure could not be found.
  
  lua(highlights) is concatenated from the following files:
  | [ERROR]:"C:\Users\waffle\AppData\Local\nvim-data\lazy\nvim-treesitter\queries\lua\highlights.scm", failed to load: Failed to load parser: uv_dlsym: The specified procedure could not be found.
  
- ERROR lua(locals): Failed to load parser: uv_dlsym: The specified procedure could not be found.
  
  lua(locals) is concatenated from the following files:
  | [ERROR]:"C:\Users\waffle\AppData\Local\nvim-data\lazy\nvim-treesitter\queries\lua\locals.scm", failed to load: Failed to load parser: uv_dlsym: The specified procedure could not be found.
  
- ERROR lua(folds): Failed to load parser: uv_dlsym: The specified procedure could not be found.
  
  lua(folds) is concatenated from the following files:
  | [ERROR]:"C:\Users\waffle\AppData\Local\nvim-data\lazy\nvim-treesitter\queries\lua\folds.scm", failed to load: Failed to load parser: uv_dlsym: The specified procedure could not be found.
  
- ERROR lua(indents): Failed to load parser: uv_dlsym: The specified procedure could not be found.
  
  lua(indents) is concatenated from the following files:
  | [ERROR]:"C:\Users\waffle\AppData\Local\nvim-data\lazy\nvim-treesitter\queries\lua\indents.scm", failed to load: Failed to load parser: uv_dlsym: The specified procedure could not be found.
  
- ERROR lua(injections): Failed to load parser: uv_dlsym: The specified procedure could not be found.
  
  lua(injections) is concatenated from the following files:
  | [ERROR]:"C:\Users\waffle\AppData\Local\nvim-data\lazy\nvim-treesitter\queries\lua\injections.scm", failed to load: Failed to load parser: uv_dlsym: The specified procedure could not be found.
 
```        

What exactly do I set it as when trying mingw_x64, is it just gcc or require('nvim-treesitter.install').compilers = { "mingw_x64" } ?

Thanks !

**Comment by clason** [comment]
Followup to https://github.com/nvim-treesitter/nvim-treesitter/issues/5216.

You need to set up your paths/development environment correctly. (Anecdotal evidence has `zig` showing the most success. Try that.)

**Comment by leet0rz** [comment]
> Followup to #5216.
> 
> You need to set up your paths/development environment correctly. (Anecdotal evidence has `zig` showing the most success. Try that.)

Zig is shown above, and zig runs just fine in shell when i type zig so I guess the path is fine for it if that is what you mean?
Thanks.

**Comment by clason** [comment]
But it's not used; you need to `require('nvim-treesitter.install').compilers = { "zig" }` _before_  calling `setup`.

**Comment by leet0rz** [comment]
> But it's not used; you need to `require('nvim-treesitter.install').compilers = { "zig" }` _before_ calling `setup`.

Yeah, it is before setup.  The selecting of zig is above and setup is below.

**Comment by leet0rz** [comment]
> Did you reinstall the parsers afterwards?

Nice, now it works. So I guess to fix my issue was the following:
1. Scoop or choco install zig (can't remember which one I Did)
2. Delete all parsers from "C:\Users\USERNAME\AppData\Local\nvim-data\lazy\nvim-treesitter\parser" folder.
3. Set: 
`require('nvim-treesitter.install').compilers = { "zig" }` above the default: `require('nvim-treesitter.configs').setup {` line in my plugins.lua
4. open some lua file again from terminal, let it re-install the parsers and then it worked.

At least that is what I did, hopefully with a lot of people having these errors it might help them in the future. Setting it to Clang or gcc as in the troubleshooting did not work for me, zig is what did it. I guess the `:TSUpdate` command is similar to what I did deleting everything in the parser folder?

Thanks for the guidance, kind of a mess but finally treesitter works.
